### PR TITLE
RequestServiceImpl and CmoRequestEntity updated

### PR DESCRIPTION
**Pull request by divyamadala30**

### CODE REVIEW

File: service/src/main/java/org/mskcc/cmo/metadb/service/CmoRequestService.java [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import org.mskcc.cmo.metadb.model.SampleManifestEntity;
 
 /**
  *
  * @author ochoaa
  */
 public interface CmoRequestService {
     void saveRequest(CmoRequestEntity request);
+    CmoRequestEntity getIgoRequest(String request_id);
+    List<SampleManifestEntity> retrieveSampleManifestList(String requestId);
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/CmoRequestService.java:14]: suggestion: rename to `getSampleManifestList`

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/CmoRequestServiceImpl.java [hunk 2]
```diff
@@ -15,10 +18,38 @@
...
+        }
+    }
+    
+    @Override
+    public CmoRequestEntity getIgoRequest(String request_id) {
+        return cmoRequestRepository.findByRequestId(request_id);
+    }
+
+    @Override
+    public List<SampleManifestEntity> retrieveSampleManifestList(String requestId) {
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/CmoRequestServiceImpl.java:52]: same suggestion here: rename to `getSampleManifestList`

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java [hunk 3]
```diff
@@ -13,25 +10,28 @@
 import org.neo4j.ogm.typeconversion.UuidStringConverter;
 
 @NodeEntity(label = "cmo_metadb_sample_metadata")
-public class SampleManifestEntity extends SampleManifest {
+public class SampleManifestEntity {
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:13]: If this isnt going to be extending SampleManifest anymore then instead of `SampleManifestJsonEntity` we should prob have it as `SampleManifest`.. something that is going to be modeling the data itself and would let us map data fetched from the database easily to a pojo

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java [hunk 4]
```diff
@@ -13,25 +10,28 @@
...
     @Relationship(type = "PX_TO_SP", direction = Relationship.INCOMING)
     private PatientMetadata patient;
     @Relationship(type = "SAMPLE_MANIFEST", direction = Relationship.OUTGOING)
-    private SampleManifest sampleManifest;
+    private SampleManifestJsonEntity sampleManifestJsonEntity;
 
     public SampleManifestEntity() {
-        super();
+        //super();
     }
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:28]: we need to clean up here instead of commenting stuff out

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java [hunk 5]
```diff
@@ -143,9 +145,10 @@ public SampleManifestEntity(UUID uuid, String mrn, String cmoPatientId, String c
             tubeId,
             cfDNA2dBarcode);
         this.uuid = uuid;
-        this.sampleList = sampleList;
+        //this.sampleList = sampleList;
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:148]: same comment here

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java [hunk 6]
```diff
@@ -155,31 +158,36 @@ public void setUuid(UUID uuid) {
...
-
     public PatientMetadata getPatient() {
         return patient;
     }
 
     public void setPatient(PatientMetadata patient) {
         this.patient = patient;
     }
+    
+    public void setSampleIgoId(Sample s) {
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:169]: So the reason why `SampleManifestEntity` was extending `SampleManifest` in the first place was because of our need for these members of `SampleManifest` and to reduce redundancy in the code.

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntityToSampleManifest.java [hunk 7]
```diff
@@ -31,12 +31,12 @@ public void setSampleManifestEntity(SampleManifestEntity sampleManifestEntity) {
         this.sampleManifestEntity = sampleManifestEntity;
     }
 
-    public SampleManifest getSampleManifest() {
-        return sampleManifest;
+    public SampleManifestJsonEntity getSampleManifest() {
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntityToSampleManifest.java:34]: the getter method name should match the field it corresponds to (`getSampleManifestJsonEntity`)

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntityToSampleManifest.java [hunk 8]
```diff
@@ -31,12 +31,12 @@ public void setSampleManifestEntity(SampleManifestEntity sampleManifestEntity) {
...
+        return sampleManifestJsonEntity;
     }
 
-    public void setSampleManifest(SampleManifest sampleManifest) {
-        this.sampleManifest = sampleManifest;
+    public void setSampleManifest(SampleManifestJsonEntity sampleManifest) {
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntityToSampleManifest.java:38]: same comment here

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestJsonEntity.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+@NodeEntity(label = "sample_manifest_json")
+public class SampleManifestJsonEntity implements Serializable {
+    @Id @GeneratedValue
+    private Long id;
+    private String sampleManifestJson;
+    private String creationDate;
+    
+    public SampleManifestJsonEntity() {}
+    
+    public SampleManifestJsonEntity(String json, String creationDate) {
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestJsonEntity.java:17]: minor: json --> sampleManifestJson for consistency

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestJsonEntity.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        this.sampleManifestJson = json;
+        this.creationDate = creationDate;
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestJsonEntity.java:19]: can you also order the getters and setters in the same way that they are listed above? (id, sampleManifestJson, creationDate)

File: persistence/src/main/java/org/mskcc/cmo/metadb/persistence/CmoRequestRepository.java [hunk 11]
```diff
@@ -12,6 +14,16 @@
  */
 @Repository
 public interface CmoRequestRepository extends Neo4jRepository<CmoRequestEntity, Long> {
-    @Query("CREATE (n:cmo_metadb_request {requestId: $requestId})")
-    public void saveNewCmoRequest(@Param("requestId") String requestId);
+    @Query("MATCH (r:cmo_metadb_request {requestId:$reqId}) RETURN r;")
```
Comment (ao508) [inline persistence/src/main/java/org/mskcc/cmo/metadb/persistence/CmoRequestRepository.java:17]: would you mind adjusting the white space to match the format from the method we had before? 

so `requestId:$reqId` -->`requestId: $reqId` for example

and apply wherever necessary below as well to keep things a bit easier to read/clean

File: persistence/src/main/java/org/mskcc/cmo/metadb/persistence/SampleManifestRepository.java [hunk 12]
```diff
@@ -14,29 +15,18 @@
...
-                + "MERGE (s:sample {sampleId:n_sample.sampleId, idSource:n_sample.idSource}) "
-                + "MERGE (s)-[:SP_TO_SP]->(sm) "
-            + ") "
-        + "MERGE (pm:cmo_metadb_patient_metadata "
-            + "{investigatorPatientId: $sample.patient.investigatorPatientId}) "
-        + "MERGE (pm)-[:PX_TO_SP]->(sm)"
-        + "RETURN sm"
-    )
-    SampleManifestEntity saveSampleManifest(@Param("sample") SampleManifestEntity sample);
+    @Query("MATCH (s:sample {value:'added-sample-divya1'}) "
```
Comment (ao508) [inline persistence/src/main/java/org/mskcc/cmo/metadb/persistence/SampleManifestRepository.java:18]: let's remove hardcoded values

File: server/src/main/java/org/mskcc/cmo/metadb/MetadbApp.java [hunk 13]
```diff
@@ -41,7 +44,8 @@ public void run(String... args) throws Exception {
             messagingGateway.connect();
             messageHandlingService.initialize(messagingGateway);
             System.out.println("Attempting to persist mock request data to neo4j..");
-            requestService.saveRequest(mockRequestData(args[0]));
+            //CmoRequestEntity req = mockRequestData("divya");
```
Comment (ao508) [inline server/src/main/java/org/mskcc/cmo/metadb/MetadbApp.java:47]: clean up instead of commenting out

File: server/src/main/java/org/mskcc/cmo/metadb/MetadbApp.java [hunk 14]
```diff
@@ -70,22 +74,36 @@ public void run() {
     public static void main(String[] args) {
         SpringApplication.run(MetadbApp.class, args);
     }
-
+    /*
     private CmoRequestEntity mockRequestData(String identifierValue) {
         CmoRequestEntity request = new CmoRequestEntity();
```
Comment (ao508) [inline server/src/main/java/org/mskcc/cmo/metadb/MetadbApp.java:79]: same comment here

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java [hunk 15]
```diff
@@ -16,6 +16,7 @@
     public SampleManifestEntity saveSampleManifest(SampleManifestEntity sample) {
         // create prerequisite request, patient nodes with the help of other services
         // sample, metadata, nodes etc
-        return sampleManifestRepository.saveSampleManifest(sample);
+        //return sampleManifestRepository.saveSampleManifest(sample);
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java:19]: this should probably not be commented out anymore

**Review by ao508 — CHANGES_REQUESTED** [review]

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java (hunk 3 above)
Comment (ao508) [reply model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:13]: thought about this a bit more, let's:

1. keep the `extends SampleManifest` 
2. remove the fields as well as the getters/setters that you added below for the [`sampleIgoId`](https://github.com/mskcc/cmo-metadb/pull/53/files#diff-f85fc40803e21bbb16de2de8850b9e536810ccc76a0247bf8ca41bb42a06416bR18) and [`sampleInvestigatorId`](https://github.com/mskcc/cmo-metadb/pull/53/files#diff-f85fc40803e21bbb16de2de8850b9e536810ccc76a0247bf8ca41bb42a06416bR20)
3. add back the [`List<Sample> sampleList`](https://github.com/mskcc/cmo-metadb/pull/53/files#diff-f85fc40803e21bbb16de2de8850b9e536810ccc76a0247bf8ca41bb42a06416bL21) as well as its getter/setter
4. add back this [default constructor ](https://github.com/mskcc/cmo-metadb/pull/53/files#diff-f85fc40803e21bbb16de2de8850b9e536810ccc76a0247bf8ca41bb42a06416bL27)

Basically what this will amount to is restoring the file to what it was before but still keep the addition of the `SampleManifestJsonEntity` that you made as well as its getter/setter

File: model/src/main/java/org/mskcc/cmo/metadb/model/CmoRequestEntity.java [hunk 16]
```diff
@@ -55,5 +56,15 @@ public String getRequestJson() {
...
     }
-
+    
+    /**
+     * 
+     * @param sampleManifestEntity
+     */
+    public void addSampleManifest(SampleManifestEntity sampleManifestEntity) {
+        if (sampleManifestList == null) {
+            sampleManifestList = new ArrayList<SampleManifestEntity>();
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/CmoRequestEntity.java:66]: You shouldnt need the type in the carrot brackets in java 8+, let's remove to keep consistent with the styling elsewhere in the code

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestJsonEntity.java [hunk 17]
```diff
@@ -0,0 +1,45 @@
+package org.mskcc.cmo.metadb.model;
+
+import java.io.Serializable;
+import org.neo4j.ogm.annotation.GeneratedValue;
+import org.neo4j.ogm.annotation.Id;
+import org.neo4j.ogm.annotation.NodeEntity;
+
+@NodeEntity(label = "sample_manifest_json")
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestJsonEntity.java:8]: let's rename the label "cmo_metadb_sample_manifest_json"

File: persistence/src/main/java/org/mskcc/cmo/metadb/persistence/SampleManifestRepository.java [hunk 18]
```diff
@@ -14,29 +16,29 @@
...
-            + "{investigatorPatientId: $sample.patient.investigatorPatientId}) "
-        + "MERGE (pm)-[:PX_TO_SP]->(sm)"
-        + "RETURN sm"
-    )
-    SampleManifestEntity saveSampleManifest(@Param("sample") SampleManifestEntity sample);
+    @Query("MATCH (s: sample {value: $igoId.sampleId, idSource: 'igoId'}) "
+        + "MATCH (s)-[:SP_TO_SP]->(sm) "
+        + "RETURN sm")
+    SampleManifestEntity findSampleByIgoId(@Param("igoId") Sample igoId);
+
```
Comment (ao508) [inline persistence/src/main/java/org/mskcc/cmo/metadb/persistence/SampleManifestRepository.java:23]: let's remove this trailing whitespace. You can do this really easily in vim with this command `:%s/\s\+$//e`

this command will remove trailing whitespace from every line in the file

**Review by ao508 — CHANGES_REQUESTED** [review]
Mostly minor changes requested but great work!

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java [hunk 19]
```diff
@@ -22,12 +21,10 @@
...
     private PatientMetadata patient;
     @Relationship(type = "SAMPLE_MANIFEST", direction = Relationship.OUTGOING)
-    private SampleManifest sampleManifest;
-
-    public SampleManifestEntity() {
-        super();
-    }
+    private SampleManifestJsonEntity sampleManifestJsonEntity;
 
+    public SampleManifestEntity() {}
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:26]: can you add the `super()` back?

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java [hunk 20]
```diff
@@ -154,32 +151,32 @@ public UUID getUuid() {
...
+    
     public void setSampleList(List<Sample> sampleList) {
         this.sampleList = sampleList;
     }
-
-    /**
-     * Add sample to array.
-     * @param sample
-     */
-    public void addSample(Sample sample) {
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:170]: can we also add back this `addSample` method?

**Review by ao508 — CHANGES_REQUESTED** [review]
2 more small adjustments then we can rebase and merge

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java (hunk 19 above)
Comment (ao508) [reply model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:26]: the default constructor is still missing  `super()`, once this is back we can rebase and merge

**Review by ao508 — CHANGES_REQUESTED** [review]

File: model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java (hunk 20 above)
Comment (ao508) [reply model/src/main/java/org/mskcc/cmo/metadb/model/SampleManifestEntity.java:170]: @divyamadala30 I dont think this has been added back yet

**Review by ao508 — APPROVED** [review]
